### DISCUSSION TITLE
Sleep on Shortcode Example (similar to using an "alias")

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#1088] [Share]
To run the script quickly, from the main prompt type:

s, then l, then space, then y to confirm.

So these four characters:
```
sl y
```

Using the `//Shortcode: ` metadata will run the script when you hit the "spacebar" after a shortcode/alias.

[Open sleep-on-shortcode in Script Kit](https://scriptkit.com/api/new?name=sleep-on-shortcode&url=https://gist.githubusercontent.com/johnlindquist/35e843a00efebb5cceeeb65ea45eb779/raw/e83dd71468ff034c9d5f288387457438d1492c38/sleep-on-shortcode.ts")

```js
// Name: Sleep on Shortcode
// Shortcode: sl

import "@johnlindquist/kit"

let confirm = await arg({
  placeholder: `Sleep system?`,
// Script Kit parses hints and assigns single key shortcuts to single letters inside of []
  hint: `[y]/[n]`,
})

if (confirm === "y") {
  sleep()
}

```

> Sidenote: From the main menu, you can also type `-` to bring up system commands.

**Comment by acairns** [comment]
The analog for me is how Spotlight, Raycast and Alfred works.

When we type in Script Kit, it matches the list of our scripts - it would be great if an alias could be added which adds it to this list.
Trying to create a script called "Sleep", to make this work, gives an error because it's the same as a system command.

But, my muscle memory is heavily built up over the years of typing "sl" and having "sleep" showing up as the first option. I hit enter at this point - and my system goes to sleep. The same goes for "emp" which searches the list of commands and matches "empty trash".

If an alias, or short code, could match on the main list when we are searching for scripts - that'd be ideal.

This would allow us to better match what we're used to in other tools, but to also avoid the "sleep is a system command" issue.

**Reply by johnlindquist** [reply]
@acairns I'll make it so matching a shortcode brings it up to "index 0" in the main list.

I'll add a metadata called "// Alias" too.

I can probably sneak these into the build I'm releasing today

**Reply by acairns** [reply]
Awesome, sounds perfect.

> I can probably sneak these into the build I'm releasing today

Oh goodness, don't tease me! :)

**Reply by johnlindquist** [reply]
@acairns Alright, snuck in alias for you 🙂

https://github.com/johnlindquist/kit/discussions/1093
